adaptive_navigation/lib/src/adaptive_navigation_scaffold.dart

```diff
@@ -183,15 +180,14 @@ class AdaptiveNavigationScaffold extends StatelessWidget {
     return Drawer(
       child: ListView(
         children: [
-          if (drawerHeader != null) drawerHeader,
+          // TODO: Find a better way to write `drawerHeader!`
+          if (drawerHeader != null) drawerHeader!,
           for (int i = 0; i < destinations.length; i++)
             ListTile(
               leading: Icon(destinations[i].icon),
               title: Text(destinations[i].title),
               onTap: () {
-                if (onDestinationSelected != null) {
-                  onDestinationSelected(i);
-                }
+                onDestinationSelected?.call(i);
```

Comment:
can you `onDestinationSelected!(i)`?

Reply:
It passes the analyzer, but if `onDestinationSelected` is null, it raises an error.

The correct logic is: call `onDestinationSelected(i)` if `onDestinationSelected != null`, otherwise do nothing. This is the same as `onDestinationSelected?.call(i)`.

Reply:
thanks, I didn't know that!